code/drasil-code/Language/Drasil/Code/Imperative/LanguageRenderer.hs

```diff
@@ -184,8 +184,8 @@ constructDocD _ = empty
 
 -- Parameters --
 
-stateParamDocD :: Label -> (Doc, CodeType) -> Doc
-stateParamDocD n (t, _) = t <+> text n
+stateParamDocD :: Label -> TypeData -> Doc
+stateParamDocD n (TD _ t) = t <+> text n
```

Comment:
might be even better to not pattern-match here and instead use the projector, so that if things get added to the data type, this code wouldn't have to change. Same with the other places where you pattern match on `TD`.

Reply:
@JacquesCarette Done for all of the cases of pattern matching except where I was pattern-matching on a specific type... I suppose I could use a `case` statement for those scenarios, but is that still preferred over pattern matching?

Reply:
No, pattern matching is better than `case` in almost all situations.